### DISCUSSION TITLE
Implementation of an API to get the device ID/name

### DISCUSSION THREAD

**Discussion by cozimus** [beeware/toga#2469] [Ideas]
I was wondering if it could be useful to add an API to get the current device ID or device name (if it's still doens't exist).
e.g. for windows you could use:

- `platform.node()` for the device name
- `uuid.getnode()` for the device ID
And those two should also work on Linux (sadly, I can't test them)

For android I thought at some implementation of:

- `android.os.Build.MODEL` for the device name. I believe briefcase already has it implemented, since it shows it when you run `briefcase run android`, but I may get it wrong
- `Settings.Secure.ANDROID_ID` for device ID

No clue about how to implement on macOS and iOS.
Any advice? Do you think it should be useful?
Thanks to everyone :)

**Comment by freakboy3742** [comment]
Thanks for the suggestion; however, I don't think there's anything extra required here in Toga. 

As part of the work on [PEP 730](https://github.com/python/cpython/issues/114099) and [PEP 738](https://github.com/python/cpython/issues/116622), you'll be able to get device detail either as part of `platform.node()`, or via the `platform.ios_ver()` and `platform.android_ver()` methods. This information is already available on macOS and Linux, with `macOS also providing `platform.mac_ver()`.

**Reply by cozimus** [reply]
That would definitively do the job, I'm looking forward to use them in the future! :)
